pages/account/[[...index]].tsx

```diff
@@ -134,9 +109,11 @@ export async function getServerSideProps({ req, res }) {
   const tabsPath = match ? match[1] : null
   const id = patronTokenResponse.decodedPatron.sub
   try {
-    const client = await sierraClient()
-    const { checkouts, holds, patron, fines, pickupLocations } =
-      await MyAccountFactory(id, client)
+    const patronData = await fetchPatronData(id)
+    const { checkouts, holds, patron, fines, pickupLocations } = JSON.parse(
```

Comment:
Hmm I don't like that you have to do JSON.parse to get the data you need. Ideally `fetchPatronData` would return the data directly and it would call `JSON.parse` internally. In this case, I'm not sure if you should set the expectation for the caller to use JSON.parse so perhaps this can be reverted. I was originally thinking that 
```
const client = await sierraClient()
    const { checkouts, holds, patron, fines, pickupLocations } =
      await MyAccountFactory(id, client)
```

would be a small function itself, but because it's two lines, maybe it's not worth it.

Reply:
updated the method!